Algo/src/roots.cxx

```diff
@@ -114,22 +104,50 @@ extern "C" void Test_roots(CCTK_ARGUMENTS) {
     CCTK_VINFO("Test_schroder succeeded in %d iterations", iters);
   }
 
+  amrex::Gpu::Buffer<CCTK_INT> niters({0});
+  auto* pniters = niters.data();
+  const amrex::Box box(amrex::IntVect(0), amrex::IntVect(0, 0, 0));
+
   {
-    const int minbits =
-        static_cast<int>(0.6 * std::numeric_limits<double>::digits);
-    const int maxiters = 100;
-    int iters;
-    bool failed;
-    auto x = newton_raphson_nd(gnd<CCTK_REAL>, vec<CCTK_REAL, 2>{1.0, 1.0},
-                               vec<CCTK_REAL, 2>{0.0, 0.0},
-                               vec<CCTK_REAL, 2>{10.0, 10.0}, minbits, maxiters,
-                               iters, failed);
-    // CCTK_VINFO("maxiters=%d iters=%d", maxiters, iters);
-    // CCTK_VINFO("lo=%.17g hi=%.17g", double(lo), double(hi));
-    assert(iters < maxiters);
-    assert(sumabs(gn(x)) < 1.0e-9);
-    CCTK_VINFO("Test_newton_raphson_nd succeeded in %d iterations", iters);
+    amrex::launch(box, [=] CCTK_DEVICE(amrex::Box const& tbx)
```

Comment:
I think that launching a GPU kernel does not wait for the kernel to finish running. Can you check, and add the respective wait statement if necessary?

Reply:
there is an implicit sync in `copyToHost`
```
    T* copyToHost ()
    {
#ifdef AMREX_USE_GPU
        if (d_data)
        {
            Gpu::dtoh_memcpy_async(h_data, d_data, m_size*sizeof(T));
            Gpu::streamSynchronize();
        }
#endif
        return h_data;
    }
```